subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/timeout/JavaProcessStackTracesMonitor.java

```diff
@@ -241,13 +245,13 @@ private static ByteArrayOutputStream connectStream(InputStream forkedProcessOutp
     }
 
 
-    public static File printAllStackTracesByJstack(File dir) {
+    public static File printAllStackTracesByJstack(File outputDir) {
         try {
             String timestamp = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd-HH-mm-ss"));
-            String output = ps().getSuspiciousDaemons().stream().map(JavaProcessInfo::jstack).collect(Collectors.joining("\n"));
-            Path outputFile = dir.toPath().resolve(timestamp + ".threaddump");
-            Files.write(outputFile, output.getBytes(StandardCharsets.UTF_8));
-            return outputFile.toFile();
+            File outputFile = new File(outputDir, timestamp + ".threaddump");
```

Comment:
What is the difference here? A refactoring?

Reply:
The point is line 252, all outputs are printed to file now.